gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/shared_policy_group/SharedPolicyGroupHistoryQueryServiceImpl.java

```diff
@@ -101,4 +102,29 @@ public Page<SharedPolicyGroup> search(String environmentId, String sharedPolicyG
             );
         }
     }
+
+    @Override
+    public Optional<SharedPolicyGroup> getLatestBySharedPolicyGroupId(String environmentId, String sharedPolicyGroupId) {
+        Assert.notNull(environmentId, "EnvironmentId must not be null");
+        Assert.notNull(sharedPolicyGroupId, "SharedPolicyGroupId must not be null");
+        try {
+            return sharedPolicyGroupHistoryRepository
+                .getLatestBySharedPolicyGroupId(environmentId, sharedPolicyGroupId)
+                .map(sharedPolicyGroupAdapter::toEntity);
+        } catch (TechnicalException e) {
+            logger.error(
+                "An error occurred while getting the latest shared policy group by environment ID {} and sharedPolicyGroupId {}",
+                environmentId,
+                sharedPolicyGroupId,
+                e
+            );
+            throw new TechnicalDomainException(
+                "An error occurred while trying to get the latest shared policy group by environment ID: " +
+                environmentId +
+                " and sharedPolicyGroupId: " +
+                sharedPolicyGroupId +
+                e
```

Comment:
Remember also to pass `e` as `cause` argument to an Exception.

Reply:
nice catch i change it